apps/emqx_bridge_iceberg/test/emqx_bridge_iceberg_SUITE.erl

```diff
@@ -0,0 +1,534 @@
+                <<"type">> => <<"int">>,
+                <<"required">> => false
+            }
+        ]
+    }.
+
+simple_setup_table(TestCase) ->
+    Client = make_client(),
+    TestCaseBin = atom_to_binary(TestCase),
+    Namespace = [TestCaseBin],
+    Table = TestCaseBin,
+    Schema = simple_schema2(),
+    ok = ensure_namespace_created(Client, Namespace),
+    ok = ensure_table_created(Client, Namespace, Table, Schema, _ExtraOpts = #{}),
+    #{ns => join_ns(Namespace), table => Table}.
+
+scan_table(Namespace, Table) ->
+    Method = get,
+    URI = iolist_to_binary(
+        lists:join(
+            "/",
+            [
+                ?QUERY_ENDPOINT,
+                "scan",
+                Namespace,
+                Table
+            ]
+        )
+    ),
+    {ok, {{_, 200, _}, _, Body}} = httpc:request(Method, {URI, []}, [], [{body_format, binary}]),
+    emqx_utils_json:decode(Body).
+
+gen_long() ->
+    <<N:32>> = crypto:strong_rand_bytes(4),
+    N.
+
+render(Template, Context) ->
+    Parsed = emqx_template:parse(Template),
+    iolist_to_binary(emqx_template:render_strict(Parsed, Context)).
+
+create_connector_api(Config, Overrides) ->
+    emqx_bridge_v2_testlib:simplify_result(
+        emqx_bridge_v2_testlib:create_connector_api(Config, Overrides)
+    ).
+
+create_action_api(Config, Overrides) ->
+    emqx_bridge_v2_testlib:simplify_result(
+        emqx_bridge_v2_testlib:create_action_api(Config, Overrides)
+    ).
+
+delete_action_api(Config) ->
+    Name = emqx_bridge_v2_testlib:get_value(action_name, Config),
+    emqx_bridge_v2_testlib:delete_kind_api(action, ?ACTION_TYPE, Name).
+
+%%------------------------------------------------------------------------------
+%% Test cases
+%%------------------------------------------------------------------------------
+
+t_start_stop() ->
+    [{matrix, true}].
+t_start_stop(matrix) ->
+    [[s3tables]];
+t_start_stop(Config) when is_list(Config) ->
+    emqx_bridge_v2_testlib:t_start_stop(Config, "iceberg_connector_stop").
+
+t_rule_action() ->
+    [{matrix, true}].
+t_rule_action(matrix) ->
+    [[s3tables]];
+t_rule_action(Config) when is_list(Config) ->
+    ct:timetrap({seconds, 15}),
+    Ns = emqx_bridge_v2_testlib:get_value(namespace, Config),
+    Table = emqx_bridge_v2_testlib:get_value(table, Config),
+    RuleTopic = atom_to_binary(?FUNCTION_NAME),
+    PayloadFn = fun() ->
+        #{
+            <<"str">> => emqx_guid:to_hexstr(emqx_guid:gen()),
+            <<"long">> => 12345678910,
+            <<"int">> => 4321,
+            %% <<"decimal">> => <<"12345678.90">>
+            <<"fixed">> => <<"123456789ABCDEF0">>
+        }
+    end,
+    NumPayloads = 3,
+    PublishFn = fun(Context) ->
+        Payloads = lists:map(fun(_) -> PayloadFn() end, lists:seq(1, NumPayloads)),
+        emqx_utils:pforeach(
+            fun(P0) ->
+                {ok, C} = emqtt:start_link(#{clean_start => true, proto_ver => v5}),
```

Comment:
We might have a bug in either `emqtt` or `emqx` regarding clientid generation/validation, respectively.

If I don't specify protocol version 5, I consistently get `client_identifier_not_valid` here...